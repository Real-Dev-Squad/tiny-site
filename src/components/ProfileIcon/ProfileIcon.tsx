import React from 'react';

interface ProfileIconProps {
    firstName: string;
    lastName: string;
    size: number;
}

const ProfileIcon: React.FC<ProfileIconProps> = ({ firstName, lastName, size }) => {
    const initials = (firstName[0] + lastName[0]).toUpperCase();
    const randomColor = Math.floor(Math.random() * 16777215).toString(16);
    const r = parseInt(randomColor.substr(0, 2), 16);
    const g = parseInt(randomColor.substr(2, 2), 16);
    const b = parseInt(randomColor.substr(4, 2), 16);
    const textColor = r * 0.299 + g * 0.587 + b * 0.114 > 186 ? 'black' : 'white';

    const styles = {
        width: size,
        height: size,
        borderRadius: '50%',
        backgroundColor: '#' + randomColor,
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        color: textColor,
        fontSize: size / 2,
    };

    return <div style={styles}>{initials}</div>;
};

export default ProfileIcon;
