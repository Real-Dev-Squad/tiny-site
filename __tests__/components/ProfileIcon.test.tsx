import React from 'react';
import { render } from '@testing-library/react';
import ProfileIcon from '@/components/ProfileIcon/ProfileIcon';

describe('ProfileIcon', () => {
    it('should have the correct initials', () => {
        const { container } = render(<ProfileIcon firstName="Sunny" lastName="Sahsi" size={50} />);
        expect(container).toHaveTextContent('SS');
    });
});
