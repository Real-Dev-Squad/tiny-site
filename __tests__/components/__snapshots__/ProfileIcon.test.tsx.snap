// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileIcon should render 1`] = `
<div>
  <div
    style="width: 50px; height: 50px; border-radius: 50%; background-color: rgb(74, 157, 39); display: flex; align-items: center; justify-content: center; color: white; font-size: 25px;"
  >
    SS
  </div>
</div>
`;
